### PR TITLE
alert dialog component

### LINKED ISSUE #2313: Introduce an Alert dialog component
**AC:**

- [ ] Create new Alert/Confirmation dialog component that uses the existing dialog
- [ ] Use this new component inside `useConfirmationDialog` hook

The goal of this task is to introduce an Alert dialog component based on the designs and docs which [can be found in Figma](https://www.figma.com/file/Pf31tuYpozYmpq163U2ho8/Web-components?node-id=6261%3A134667). Please note, that the Alert dialog component is just an encapsulation of the Dialog component with predefined properties, contents and usage guidelines on top of it.

image

### PR DESCRIPTION
**Pull request by rafalpawlow**
Closes #2313 

**Design**
https://www.figma.com/file/Pf31tuYpozYmpq163U2ho8/Web-components?node-id=6278%3A138198

### CODE REVIEW

File: packages/atlas/src/components/_overlays/Dialog/Dialog.styles.ts [hunk 1]
```diff
@@ -122,7 +122,26 @@ export const FooterButtonsContainer = styled.div`
...
+export type InformativeIconVariant = 'destructive' | 'warning' | 'primary'
+
+const getInformativeIconColor = (variant?: InformativeIconVariant) => {
+  if (variant === 'destructive') {
+    return cVar('colorCoreRed400')
+  }
+  if (variant === 'warning') {
+    return cVar('colorCoreYellow200')
+  }
+  return cVar('colorCoreBlue500')
```
Comment (eldiegod) [inline packages/atlas/src/components/_overlays/Dialog/Dialog.styles.ts]: should add the variant informative with icon color Neutral 100

File: packages/atlas/src/components/_overlays/Dialog/Dialog.tsx [hunk 2]
```diff
@@ -41,7 +47,15 @@ export type DialogProps = {
   children?: React.ReactNode
   as?: React.ElementType
   onSubmit?: (e?: FormEvent) => void
-}
+} & (
+  | { variant?: 'default'; iconType?: DialogIconType; type?: undefined; size?: DialogSize }
+  | {
+      variant?: 'alert'
+      iconType?: undefined
+      type?: 'destructive' | 'warning' | 'primary'
```
Comment (eldiegod) [inline packages/atlas/src/components/_overlays/Dialog/Dialog.tsx]: `informative` instead of `primary`?

**Review by eldiegod — CHANGES_REQUESTED** [review]

**Review by eldiegod — APPROVED** [review]

**Review by kdembler — COMMENTED** [review]
During grooming we discussed that we wanted to have this as a separate component, that's also what is listed in the AC, what's the reason to keep all this functionality inside `Dialog`?

**Comment by rafalpawlow** [comment]
@kdembler I misunderstood AC, will fix it

**Comment by kdembler** [comment]
@rafalpawlow Do you agree it makes sense to extract this into a separate component? I think this way we end up with simpler `Dialog` which we can compose into additional components

**Comment by rafalpawlow** [comment]
@kdembler yes, I totally agree

**Comment by eldiegod** [comment]
I'm a bit scared about this PR clashing too hard with mine on https://github.com/Joystream/atlas/pull/2462

**Review by kdembler — APPROVED** [review]

**Pull request merged**